### ISSUE TITLE
Request for access for `awkward`, `awkward-cpp`, and `uproot`

### ISSUE THREAD

**Issue by agoose77** [scientific-python/upload-nightly-action#29]
We (over at scikit-hep) maintain several packages that would benefit from nightlies. To begin with, I'd like to request access for the following projects:

- `awkward`
- `awkward-cpp`
- `uproot`

Thanks all!

**Comment by matthewfeickert** [comment]
Following up on the discussions started in https://github.com/scientific-python/upload-nightly-action/issues/20#issuecomment-1594574923 I think that further approval / review requires people with org level admin powers, so tagging @tupui, @stefanv, and @jarrodmillman. (I think there may be others, but I don't know.)

**Comment by pllim** [comment]
I had also once asked if `astropy` should moved its nightlies to SP. (We already host it somewhere else.) The motivation is that downstream can pull the whole stack from the same index URL. But if not, also no big deal. I guess it depends on the goal and scope of SP hosting.

**Comment by matthewfeickert** [comment]
@pllim to avoid crossover of discussions can I ask that you make a request in a separate Issue like Issue #20 and Issue #29?

https://github.com/scientific-python/upload-nightly-action/blob/a2ba178f650e68bd6f9d8e2b7d59d16ffd257c29/README.md?plain=1#L39-L45

(We don't have this super formalized at the moment, but c.f. https://github.com/scientific-python/upload-nightly-action/issues/20#issuecomment-1594574923 for how we might make it so in the future.)

**Comment by matthewfeickert** [comment]
@agoose77 as I know that `awkward` uses `hatchling`, if your request is approved just a heads up that you might run into https://github.com/Anaconda-Platform/anaconda-client/issues/676. I will tag Ofek there to see if he has any suggestions for the `anaconda-client` team.

**Comment by jarrodmillman** [comment]
I am +1 for adding 
- `awkward`
- `awkward-cpp`
- `uproot`
- `astropy` nightlies

We probably want to add some more guidance about what we can include, but we can sort things out as we go. I am busy this weekend, but can look into this next week.

**Comment by bsipocz** [comment]
> We probably want to add some more guidance about what we can include, but we can sort things out as we go.

I really would prefer if the guidelines are figured out first than starting to add random packages (==non core, and core dependencies), then ending up to need to say no for others later when we run into technical limitations, e.g. not enough space, etc.
That guideline could be very generic, that e.g. specific things from the domain stack could use the same space, etc, but I definitely see the value of figuring it out first rather than when a handful of  packages are already there.

**Comment by matthewfeickert** [comment]
To mitigate this I've made a [`scikit-hep-nightly-wheels` Anaconda Cloud organization](https://anaconda.org/scikit-hep-nightly-wheels) which with https://github.com/scikit-hep/awkward/pull/3012 is now taking care of nightly wheels for `awkward` and `awkward-cpp` (contingent on https://github.com/Anaconda-Platform/anaconda-client/issues/702 being resolved).

I'll let the Awkward dev team decide if they want to keep this Issue open for future discussion, of if they are content with this solution for the long term, in which case they can close it. :+1:

**Comment by stefanv** [comment]
Here's a somewhat radical proposal: include everyone who will find this service useful. Once we outgrow our space, we'll have to tell new projects "no" until we figure out a solution. The solution may be as easy as hosting our own conda server, for which we already have server capacity.

**Comment by martinfleis** [comment]
>  include everyone who will find this service useful

I don't find it that radical. As far as I understood this initiative when we discussed it during the summit in Seattle last year, this was the intention. I wasn't involved very much in subsequent discussions but if there's a way of providing this service to the whole ecosystem, rather than a selection of packages based on unclear criteria and essentially a _feeling_ of a package importance, let's try to do that.

**Comment by matthewfeickert** [comment]
@stefanv @martinfleis I am +1 on this. I would be very happy to accept the requests that are already open in this Issue and Issues #45, #50, and #51, but to have this be a decision that is easily understood in the future can the following happen (in the very near future)?

* Have the @scientific-python/spec-steering-committee agree on moving forward with this proposal in the very near future
* Have the @scientific-python/spec-steering-committee advise on what a future policy would be (this is https://github.com/scientific-python/upload-nightly-action/issues/30#issuecomment-1934573445) so that when space limits are hit there is a clear transition plan.

@stefanv as you are a member of the @scientific-python/spec-steering-committee can you comment on how soon you think a discussion could take place? I'm not sure how often async discussions happen, but I'd be keen to move on this.

> The solution may be as easy as hosting our own conda server, for which we already have server capacity.

@stefanv Can you comment (over on https://github.com/scientific-python/upload-nightly-action/issues/30) on how much server space Scientific Python realistically has? If more than 30 GB could be given to storage of wheels then I would be interested in helping to get this setup even if this is just as a test instance. I'm in general +1 on controlling infrastructure, but if Scientific Python's grant funds can be spent on things other than server costs I'm also +1 on taking advantage of Anaconda's generosity to open source.

**Comment by matthewfeickert** [comment]
@agoose77 @jpivarski Good news, we have the approval (https://github.com/scientific-python/upload-nightly-action/issues/30#issuecomment-1960037619) to add `awkward-cpp`, `awkward`, and `uproot` to the nightly wheels Anaconda Cloud org https://anaconda.org/scientific-python-nightly-wheels/ !

To get started on this can @agoose77 and @jpivarski please do the following:

## Steps to gain upload access to https://anaconda.org/scientific-python-nightly-wheels/

(@jpivarski already did these steps in https://github.com/scikit-hep/awkward/pull/3012 for https://anaconda.org/scikit-hep-nightly-wheels/ so he doesn't need to do them again, and the token he made is still good here. :+1: Though @agoose77 if you want upload permissions/responsibility then please make an Aanconda Cloud account too.)

* [x] ~Create an account on [Anaconda Cloud](https://anaconda.org/)~
* [x] ~Reply here with the link to your account~
* [x] ~Make a new Anaconda Cloud API token with permissions described in https://scientific-python.org/specs/spec-0004/#process-for-adding-new-projects~

![image](https://github.com/scientific-python/upload-nightly-action/assets/5142394/697ce89f-4a51-43d0-a26f-2493f4bdce34)

* [x] ~Save the API token as a repo GitHub Actions secret named `ANACONDA_ORG_UPLOAD_TOKEN`~
* [ ] Verify that I have added you to the `Awkward` and `Uproot` team "Group" on https://anaconda.org/scikit-hep-nightly-wheels

If you'd also like to use the `scientific/upload-nightly-action` GitHub Action that this repository provides please check out [this repo's README](https://github.com/scientific-python/upload-nightly-action?tab=readme-ov-file#nightly-upload). If you need to collect wheels from different CI jobs, consider creating a new GitHub Actions workflow with logic similar to [`matplotlib`'s](https://github.com/matplotlib/matplotlib/blob/88d64e5f0cc82a050b840e61ac60e1081bb6d799/.github/workflows/nightlies.yml) or [`Awkward`'s](https://github.com/scikit-hep/awkward/pull/3012).

If you have any questions please just ask here!
